code/modules/mining/remote_fob.dm

```diff
@@ -0,0 +1,329 @@
+/////////For remote construction of FOB using a computer on the ship during setup phase
+GLOBAL_LIST_INIT(dropship_lzs, typecacheof(list(/area/shuttle/drop1/lz1, /area/shuttle/drop2/lz2)))
+
+
+/////////////////////////////// Drone Mob
+
+/mob/camera/aiEye/remote/fobdrone
+	name = "Remote Construction Drone"
+	mouse_opacity = MOUSE_OPACITY_OPAQUE
+	icon = 'icons/Marine/remotefob.dmi'
+	icon_state = "drone"
+	var/area/starting_area
+	use_static = FALSE
+	var/turf/spawnloc
+
+/mob/camera/aiEye/remote/fobdrone/Initialize()
+	. = ..()
+	starting_area = get_area(loc)
+
+/mob/camera/aiEye/remote/fobdrone/setLoc(atom/t) //unrestricted movement inside the landing zone
+	var/area/curr_area = get_area(t)
+	if(istype(t, /turf/closed) || locate(/obj/machinery/door/poddoor) in t)
+		return
+	if(curr_area == starting_area || is_type_in_typecache(curr_area, GLOB.dropship_lzs))
+		return ..()
```

Comment:
The logical progress of this proc is a series of checks to see if it should go through, and calling the parent if that is the case. In that sense it's a bit more clear if these checks are negated and the call to parent happens at the end, alternatively.
In practice:
```
if(curr_area != starting_area && !is_type_in_typecache(curr_area, GLOB.dropship_lzs))
	return
return ..()
```